src/http.rs

```diff
@@ -1009,3 +1157,101 @@ impl Drop for HttpStream {
+            )*
+        )
+        => {
+            pub fn process_request(req: Request<Body>) -> Response<Body> {
+                match req.uri().path() {
+                    "/" => {
+                        serve(
+                            include_bytes!(
+                                "../contrib/routinator-ui/index.html"
+                            ),
+                            self::ctypes::html
+                        )
+                    }
+                    $(
+                        $(
+                            concat!("/ui/", $path, ".", stringify!($ext)) => {
+                                serve(
+                                    include_bytes!(
+                                        concat!(
+                                            "../contrib/routinator-ui/",
+                                            $path, ".",
+                                            stringify!($ext)
+                                        )
+                                    ),
+                                    self::ctypes::$ext
+                                )
+                            }
+                        )*
+                    )*
+                    _ => super::not_found()
+                }
+            }
+        }
+    }
+
+    assets!(
+        ("favicon" => ico),
+        ("css/app" => css),
+        ("fonts/element-icons" => ttf, woff),
+        ("fonts/lato-latin-100" => woff, woff2),
+        ("fonts/lato-latin-300" => woff, woff2),
+        ("fonts/lato-latin-400" => woff, woff2),
+        ("fonts/lato-latin-700" => woff, woff2),
+        ("fonts/lato-latin-900" => woff, woff2),
+        ("fonts/lato-latin-100italic" => woff, woff2),
+        ("fonts/lato-latin-300italic" => woff, woff2),
+        ("fonts/lato-latin-400italic" => woff, woff2),
+        ("fonts/lato-latin-700italic" => woff, woff2),
+        ("fonts/lato-latin-900italic" => woff, woff2),
+        ("fonts/source-code-pro-latin-200" => woff, woff2),
+        ("fonts/source-code-pro-latin-200" => woff, woff2),
+        ("fonts/source-code-pro-latin-300" => woff, woff2),
+        ("fonts/source-code-pro-latin-400" => woff, woff2),
+        ("fonts/source-code-pro-latin-500" => woff, woff2),
+        ("fonts/source-code-pro-latin-600" => woff, woff2),
+        ("fonts/source-code-pro-latin-700" => woff, woff2),
+        ("fonts/source-code-pro-latin-900" => woff, woff2),
+        ("img/afrinic" => svg),
+        ("img/apnic" => svg),
+        ("img/arin" => svg),
+        ("img/blue" => svg),
+        ("img/lacnic" => svg),
+        ("img/ripencc" => svg),
+        ("img/routinator_logo_white" => svg),
+        ("img/welcome" => svg),
+        ("js/app" => js),
+    );
+
+    fn serve(data: &'static [u8], ctype: &'static [u8]) -> Response<Body> {
+        Response::builder()
+        .header("Content-Type", ctype)
+        .body(data.into())
+        .unwrap()
+    }
+
+    #[allow(non_upper_case_globals)]
+    mod ctypes {
```

Comment:
minor nit: I'd call these things `mtypes` or `mime_types`. `ctypes` got me confused, thinking it had something to do with C FFI or something

Reply:
Officially, these they have been called _content types_ for, like, ever. I think this is a leftover of my SIP past for some reason. (It also is what the header is called, so there …)